iceoryx_posh/test/moduletests/test_popo_waitset.cpp

```diff
@@ -0,0 +1,318 @@
+{
+  public:
+    bool isConditionVariableAttached() noexcept override
+    {
+        return m_condVarAttached;
+    }
+
+    bool attachConditionVariable(ConditionVariableData* ConditionVariableDataPtr) noexcept override
+    {
+        m_condVarAttached = true;
+        m_condVarPtr = ConditionVariableDataPtr;
+        return m_condVarAttached;
+    }
+
+    bool hasTrigger() noexcept override
+    {
+        return m_wasTriggered;
+    }
+
+    void resetTrigger() noexcept override
+    {
+        m_wasTriggered = false;
+    }
+
+    bool detachConditionVariable() noexcept override
+    {
+        m_condVarAttached = false;
+        m_condVarPtr = nullptr;
+        return m_condVarAttached;
+    }
+
+    /// @note done in ChunkQueuePusher
+    void notify()
+    {
+        m_wasTriggered = true;
+        ConditionVariableSignaler signaler{m_condVarPtr};
+        signaler.notifyOne();
+    }
+
+    /// @note members reside in ChunkQueueData in SHM
+    bool m_condVarAttached{false};
+    bool m_wasTriggered{false};
+    ConditionVariableData* m_condVarPtr{nullptr};
+};
+
+class WaitSet_test : public Test
+{
+  public:
+    static constexpr uint16_t MAX_NUMBER_OF_CONDITIONS_WITHOUT_GUARD = iox::popo::MAX_NUMBER_OF_CONDITIONS - 1;
+
+    ConditionVariableData m_condVarData;
+    WaitSet m_sut{&m_condVarData};
+    vector<MockSubscriber, MAX_NUMBER_OF_CONDITIONS_WITHOUT_GUARD> m_subscriberVector;
+
+    iox::posix::Semaphore m_syncSemaphore = iox::posix::Semaphore::create(0u).get_value();
+
+    void SetUp()
+    {
+        MockSubscriber subscriber;
+        while (m_subscriberVector.size() != m_subscriberVector.capacity())
+        {
+            m_subscriberVector.push_back(subscriber);
+        }
+    };
+
+    void TearDown()
+    {
+        m_subscriberVector.clear();
+        m_sut.clear();
+        ConditionVariableWaiter waiter{&m_condVarData};
+        waiter.reset();
+    };
+};
+
+TEST_F(WaitSet_test, AttachSingleConditionSuccessful)
+{
+    EXPECT_TRUE(m_sut.attachCondition(m_subscriberVector.front()));
+}
+
+TEST_F(WaitSet_test, AttachSameConditionTwiceResultsInFailure)
+{
+    m_sut.attachCondition(m_subscriberVector.front());
+    EXPECT_FALSE(m_sut.attachCondition(m_subscriberVector.front()));
+}
+
+TEST_F(WaitSet_test, AttachMultipleConditionSuccessful)
+{
+    for (auto& currentSubscriber : m_subscriberVector)
+    {
+        EXPECT_TRUE(m_sut.attachCondition(currentSubscriber));
+    }
+}
+
+TEST_F(WaitSet_test, AttachTooManyConditionsResultsInFailure)
+{
+    for (auto& currentSubscriber : m_subscriberVector)
+    {
+        m_sut.attachCondition(currentSubscriber);
+    }
+
+    MockSubscriber extraCondition;
+    EXPECT_FALSE(m_sut.attachCondition(extraCondition));
+}
+
+TEST_F(WaitSet_test, DetachSingleConditionSuccessful)
+{
+    m_sut.attachCondition(m_subscriberVector.front());
+    EXPECT_TRUE(m_sut.detachCondition(m_subscriberVector.front()));
+}
+
+TEST_F(WaitSet_test, DetachMultipleleConditionsSuccessful)
```

Comment:
Typo Multiplelelelelele not Multiplele ;)

Reply:
Fixed